urlb57eb64056_jacadenac_ISI_Othello_UN_tgz-pJ8-othello_OthelloJ8/src/logica/Torneo.java

```diff
@@ -0,0 +1,131 @@
+			datosTorneo += "*Juego número "+(i+1)+":\r\n";
+			int[] estrategia2 = new int[4];
+			for( int j=0 ; j<estrategiasCompetidoras[i].length-1 ; j++){
+				estrategia2[j] = estrategiasCompetidoras[i][j];
+			}
+			
+			datosTorneo += "     Estrategia 1 = [ ";
+			for(int j=0;j<4;j++){
+				datosTorneo += ""+estrategia1[j];
+				if (j<3) datosTorneo += ", ";
+				else datosTorneo += " ]\r\n"; 
+			}
+			datosTorneo += "     Estrategia 2 = [ ";
+			for(int j=0;j<4;j++){
+				datosTorneo += ""+estrategia2[j];
+				if (j<3) datosTorneo += ", ";
+				else datosTorneo += " ]\r\n"; 
+			}
+			
+			//alterna el color del jugador con estrategia1 de acuerdo al número del torneo
+			if (numeroTorneosCreados%2 == 0 && numeroTorneosCreados%3 != 0) {
+				juego = new JuegoTorneo(estrategia1, estrategia2, TipoTablero.OCTOGONAL, tamTablero);
+			} else if (numeroTorneosCreados%3 == 0) {
+				juego = new JuegoTorneo(estrategia1, estrategia2, TipoTablero.PERSONALIZADO, tamTablero);
+			}  else {
+				juego = new JuegoTorneo(estrategia2, estrategia1, TipoTablero.CLASICO, tamTablero);
+			}
+			juego.start();
+			if(estrategia1 == juego.getGanador()){
+				datosTorneo += "     Ganó estrategia 1\r\n";
+				//System.out.println("Ganó estrategia 1");
+				cantidadVictorias++;
+			}else if(estrategia2 == juego.getGanador()){
+				datosTorneo += "     Ganó estrategia 2\r\n";
+				//System.out.println("Ganó estrategia 2");
+			}else if(juego.getGanador() == null){
+				datosTorneo += "No Ganó ninguan estrategia\r\n";
+				//System.out.println("No Ganó ninguan estrategia");
+			}
+		}
+		datosTorneo += "\r\n     Estrategia 1 obtuvo " + getCantidadVictorias() + " victorias de " 
+					   + estrategiasCompetidoras.length + ".\r\n\r\n";
+		guardarDatosTorneo(datosTorneo,"Torneos.txt");
+		datosTorneo = ""+numeroTorneosCreados;
+		guardarDatosTorneo(datosTorneo,"parametrosIniciales.txt");
+		
+	}
+	
+	public int getCantidadVictorias() {
+		return cantidadVictorias;
+	}
+	
+	public void guardarDatosTorneo(String datos, String nombreArchivo){
+		String nombreFichero = nombreArchivo;
+		if (!directorio.exists()) {
+            directorio.mkdirs();
+        }
+		try {
+			boolean bandera = true;
+			if(nombreFichero == "parametrosIniciales.txt"){
+				bandera = false;
+			}
+			FileWriter fichero = new FileWriter(directorio.getPath()+"/"+nombreFichero, bandera);
+			PrintWriter writer;
+			writer = new PrintWriter(fichero);
+			writer.println(datos);
+			writer.close();
+		} catch (IOException e) {
+			System.out.println("No se ha podido escribir en el archivo torneo.txt");
+		}
+	}
+	
+	public String getDatosTorneo(){
+		return datosTorneo;
+	}
+	
+	private static void leerParametros() {
+		JFileChooser fileChooser = new JFileChooser();
+		String nombreFichero = "parametrosIniciales.txt";
+		File fichero = new File(directorio.getPath(),nombreFichero);
+		fileChooser.setCurrentDirectory(fichero);		
+		try {
+			BufferedReader reader = new BufferedReader(new FileReader(fichero));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *reader*. The resource is referenced at lines 118, 119, 120. The resource is closed at line 123. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *reader* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).